feat: implement event parsing and storage for RecipientRecieved events

yahgw-ai opened the pull request:
## Summary
- Implemented event parsing using ethers interface to extract recipient and value from RecipientRecieved events
- Added functionality to create unique event keys using transactionHash:logIndex format  
- Build and persist RecipientRecievedEventData structure with metadata and parsed events
- Merge new events with existing data and update last_scanned_block
- Ensure all addresses are stored in checksummed format

## Test plan
- [x] All existing unit tests pass
- [x] Added comprehensive tests for event parsing and storage functionality
- [x] Tests cover event key generation, data merging, address checksumming, and last_scanned_block updates

Closes #180

yahgw-ai committed:
test: add failing test for parsing RecipientRecieved events with ethers interface

yahgw-ai committed:
feat: implement event parsing and storage for RecipientRecieved events

yahgw-ai committed:
test: update tests to match new event parsing and storage implementation

yahgwai commented on src/recipient-recieved-scanner.ts:
```diff
@@ -261,12 +263,27 @@ export class RecipientRecievedScanner {
-      void events; // Events collected but not processed yet
+      // Parse and store events (will be accumulated)
+      if (events.length > 0) {
+        await this.parseAndStoreEvents(address, events);
```
Instead of storing now as we go we should accumulate this data and store once at the end

yahgwai commented on src/recipient-recieved-scanner.ts:
```diff
@@ -292,4 +309,77 @@ export class RecipientRecievedScanner {
+      this.fileManager.readRecipientRecievedEvents(distributorAddress);
+
+    // Get chain ID from distributors data
+    const distributorsData = this.fileManager.readDistributors();
```
We've called this once before - just pass the chain id around instead of calling the filemanager again

yahgw-ai committed:
refactor: accumulate events and store once at end

- Refactor processDistributor to accumulate all events before storing
- Pass chain ID as parameter to parseAndStoreEvents to avoid redundant file reads
- Update parseAndStoreEvents signature to accept additional parameters
- Fix test mocks to include writeRecipientRecievedEvents

This addresses PR review comments to improve performance and reduce file I/O operations.

yahgwai commented on src/recipient-recieved-scanner.ts:
```diff
@@ -292,4 +317,61 @@ export class RecipientRecievedScanner {
+    chainId: number,
+    lastProcessedBlock: number,
+    existingData: RecipientRecievedEventData | undefined,
+  ): Promise<void> {
```
why is this function async? i dont think it needs to be

yahgw-ai committed:
refactor: remove unnecessary async from parseAndStoreEvents

The parseAndStoreEvents function doesn't contain any async operations,
so the async/await keywords are not needed.
